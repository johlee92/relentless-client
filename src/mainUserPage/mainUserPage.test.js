import React from 'react';
import ReactDOM from 'react-dom';
// because using Enzyme to test
// need to make sure the setupTest.js file links to adapter
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MainUserPage from './mainUserPage';
import { BrowserRouter } from 'react-router-dom';

describe(`MainUserPage component`, () => {
    const props = {
        className: 'test-class-name'
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter> 
                <MainUserPage />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the component by default', () => {
        const wrapper = shallow(<MainUserPage />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the component with props', () => {
        const wrapper = shallow(<MainUserPage {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
