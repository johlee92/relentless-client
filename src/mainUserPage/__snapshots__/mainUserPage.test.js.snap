// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainUserPage component renders the component by default 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "currentGoals": Object {
          "annualGoals": Array [],
          "monthlyGoals": Array [],
          "weeklyGoals": Array [],
        },
        "goalsFetch": [Function],
      }
    }
  >
    <PageNav />
    <div
      className="main-user-page"
    >
      <div
        className="main-user-page-container"
      >
        <GoalsNav
          changeViewGoals={[Function]}
          changeWeekNum={[Function]}
          weekNum={16}
          yearNum={2020}
        />
        <GoalsDashboard
          annual_goals={Array []}
          currentGoals={
            Object {
              "annualGoals": Array [],
              "monthlyGoals": Array [],
              "weeklyGoals": Array [],
            }
          }
          currentWeekGoals={Array []}
          monthly_goals={Array []}
          weekly_goals={Array []}
        />
        <h2>
          Period Goals:
        </h2>
        <button
          className="add-goals"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          Add Goals
        </button>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <Goals
            changeDisplayEditGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            viewGoals="weekly"
          />
        </span>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          <AddGoal
            changeDisplayAddGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            viewGoals="weekly"
            weekNum={16}
            yearNum={2020}
          />
        </span>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          <EditGoal
            changeDisplayEditGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            targetGoal=""
            targetGoalId=""
            viewGoals="weekly"
          />
        </span>
      </div>
    </div>
    <PageFooter />
  </ContextProvider>
</div>
`;

exports[`MainUserPage component renders the component with props 1`] = `
<div>
  <ContextProvider
    value={
      Object {
        "currentGoals": Object {
          "annualGoals": Array [],
          "monthlyGoals": Array [],
          "weeklyGoals": Array [],
        },
        "goalsFetch": [Function],
      }
    }
  >
    <PageNav />
    <div
      className="main-user-page"
    >
      <div
        className="main-user-page-container"
      >
        <GoalsNav
          changeViewGoals={[Function]}
          changeWeekNum={[Function]}
          weekNum={16}
          yearNum={2020}
        />
        <GoalsDashboard
          annual_goals={Array []}
          currentGoals={
            Object {
              "annualGoals": Array [],
              "monthlyGoals": Array [],
              "weeklyGoals": Array [],
            }
          }
          currentWeekGoals={Array []}
          monthly_goals={Array []}
          weekly_goals={Array []}
        />
        <h2>
          Period Goals:
        </h2>
        <button
          className="add-goals"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
        >
          Add Goals
        </button>
        <span
          style={
            Object {
              "display": "block",
            }
          }
        >
          <Goals
            changeDisplayEditGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            viewGoals="weekly"
          />
        </span>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          <AddGoal
            changeDisplayAddGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            viewGoals="weekly"
            weekNum={16}
            yearNum={2020}
          />
        </span>
        <span
          style={
            Object {
              "display": "none",
            }
          }
        >
          <EditGoal
            changeDisplayEditGoals={[Function]}
            currentGoals={
              Object {
                "annualGoals": Array [],
                "monthlyGoals": Array [],
                "weeklyGoals": Array [],
              }
            }
            targetGoal=""
            targetGoalId=""
            viewGoals="weekly"
          />
        </span>
      </div>
    </div>
    <PageFooter />
  </ContextProvider>
</div>
`;
